### PR TITLE
readUint32LE to readInt32LE deserialize

### PR DESCRIPTION
**Pull request by loucaponi**
Small change to fix version field when deserializing new transactions

### CODE REVIEW

File: src/transaction.js [hunk 1]
```diff
@@ -204,7 +204,7 @@ function deserializeTx (hexStr: string, withPrevScriptPubKey: boolean = false):
   var txObj = { version: 0, locktime: 0, ins: [], outs: [] }
 
   // Version
-  txObj.version = buf.readUInt32LE(offset)
+  txObj.version = buf.readInt32LE(offset)
```
Comment (MarcoOl94) [inline src/transaction.js:207]: It should be good, like in bitcore-lib-zen (https://github.com/HorizenOfficial/bitcore-lib-zen/blob/zendoo-supportPR/lib/transaction/transaction.js#L376) but maybe test it.

Comment (loucaponi) [reply src/transaction.js:207]: @MarcoOl94 we noticed that `readUInt32LE` breaks when we see negative version numbers (i.e. version: -4).

Do you see a better way to approach this issue?

Comment (MarcoOl94) [reply src/transaction.js:207]: Which test did you do? Can you please try to deserialize a raw Tx, serialize the result and verify that the output is equal to the initial rawTx? (both with readUInt32LE and readInt32LE)

**Review by MarcoOl94 — APPROVED** [review]

File: src/transaction.js (hunk 1 above)
Comment (MarcoOl94) [reply src/transaction.js:207]: Ok I'll approve it. Well done!

**Review by paolocappelletti — APPROVED** [review]

**Pull request merged**